src/script/ui/state/server/index.js

```diff
@@ -127,6 +142,50 @@ export function serverTransform(method, data, struct) {
 	};
 }
 
+function deleteAllSGroupsWithName(restruct, action, fieldName) {
+	restruct.molecule.sgroups.forEach((sg, id) => {
+		if (sg.data.fieldName === fieldName)
+			action.mergeWith(fromSgroupDeletion(restruct, id));
+	});
+}
+
+export function calculateCip(result) {
```

Comment:
could you add a comment / documentation on this method and deleteAllSGroupsWithName detailing what the methods do & how they're used (for this exported one).